### PR TITLE
Add google analytics tracking

### LINKED ISSUE #341: Integrate GA

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Add Google Analytics tracking
- Add GA event for filter panel visibility
- Add GA event for user panel visibility
- Add GA event for user expanding topology

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Sanity check code
-  Remove the `if (process.env.NODE_ENV === "production")` conditional
- Open demo link and verify yourself in realtime in GA dashboard

## Details

Screenshot 2020-02-12 at 12 32 27

Fixes #341

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -12,6 +13,10 @@ import NotFound from "pages/NotFound/NotFound";
 const baseURL = process.env.REACT_APP_BASE_APP_URL;
 
 function App() {
+  if (process.env.NODE_ENV === "production") {
+    ReactGA.initialize("UA-1018242-68");
+    ReactGA.pageview(window.location.pathname + window.location.search);
```
Comment (hatched) [inline src/components/App/App.js]: We probably also want the hash so a more resilient approach might be to do...Unless we don't, in which case ignore this :-)
```suggestion
    ReactGA.pageview(window.location.href.replace(window.location.origin, ''));
```

File: src/components/FilterTags/FilterTags.js [hunk 2]
```diff
@@ -145,7 +146,15 @@ const FilterTags = () => {
       <input
         type="text"
         className="p-filter-tags__input"
-        onClick={() => setFilterPanelVisibility(true)}
+        onClick={() => {
+          setFilterPanelVisibility(true);
+          if (filterPanelVisibility) {
+            ReactGA.event({
+              category: "User",
```
Comment (hatched) [inline src/components/FilterTags/FilterTags.js]: We should have a common place where we list the available categories.

File: src/components/FilterTags/FilterTags.js [hunk 3]
```diff
@@ -145,7 +146,15 @@ const FilterTags = () => {
       <input
         type="text"
         className="p-filter-tags__input"
-        onClick={() => setFilterPanelVisibility(true)}
+        onClick={() => {
+          setFilterPanelVisibility(true);
+          if (filterPanelVisibility) {
```
Comment (hatched) [inline src/components/FilterTags/FilterTags.js]: The previous line you're setting the `filterPanelVisibility` to `true` so this check is not necessary as it should always be called.

File: src/components/InfoPanel/InfoPanel.js [hunk 4]
```diff
@@ -53,7 +54,15 @@ const InfoPanel = () => {
           <Topology width={300} height={300} modelData={modelStatusData} />
           <i
             className="p-icon--expand"
-            onClick={() => setShowExpandedTopology(!showExpandedTopology)}
+            onClick={() => {
+              setShowExpandedTopology(!showExpandedTopology);
+              if (showExpandedTopology) {
```
Comment (hatched) [inline src/components/InfoPanel/InfoPanel.js]: This should be the same value as being set on 58, yes? As at this point in the code the `showExpandedTopology` will be the old value.

File: src/components/UserIcon/UserIcon.js [hunk 5]
```diff
@@ -48,7 +49,15 @@ export default function User() {
   return (
     <div
       className="user-icon"
-      onClick={() => setUserPanelVisibility(!userPanelVisibility)}
+      onClick={() => {
+        setUserPanelVisibility(!userPanelVisibility);
+        if (userPanelVisibility) {
```
Comment (hatched) [inline src/components/UserIcon/UserIcon.js]: See above comment about this value.

File: src/components/FilterTags/FilterTags.js (hunk 2 above)
Comment (barrymcgee) [reply src/components/FilterTags/FilterTags.js]: A task for further down the line

File: src/components/InfoPanel/InfoPanel.js (hunk 4 above)
Comment (barrymcgee) [reply src/components/InfoPanel/InfoPanel.js]: Because it's not in the callback, good catch.

File: src/components/InfoPanel/InfoPanel.js [hunk 6]
```diff
@@ -53,7 +54,13 @@ const InfoPanel = () => {
           <Topology width={300} height={300} modelData={modelStatusData} />
           <i
             className="p-icon--expand"
-            onClick={() => setShowExpandedTopology(!showExpandedTopology)}
+            onClick={() => {
+              setShowExpandedTopology(!showExpandedTopology);
+              ReactGA.event({
```
Comment (hatched) [inline src/components/InfoPanel/InfoPanel.js]: This is triggering that it expanded even if it's closing. You'll want to add a conditional here.

File: src/components/UserIcon/UserIcon.js [hunk 7]
```diff
@@ -45,10 +46,21 @@ export default function User() {
...
+        action: "Opened user panel"
+      });
+    }
+  }, [userPanelVisibility]);
+
   return (
     <div
       className="user-icon"
-      onClick={() => setUserPanelVisibility(!userPanelVisibility)}
+      onClick={() => {
```
Comment (hatched) [inline src/components/UserIcon/UserIcon.js]: You can remove the brackets here.

File: src/components/InfoPanel/InfoPanel.js (hunk 6 above)
Comment (barrymcgee) [reply src/components/InfoPanel/InfoPanel.js]: Negative - the close button is different.

**Pull request merged**